server/pulp/server/db/manage.py

```diff
@@ -165,7 +165,16 @@ def ensure_database_indexes():
 
     # Load all the model classes that the server knows about and ensure their indexes as well
     plugin_manager = PluginManager()
-    for model_class in plugin_manager.unit_models.itervalues():
+    for unit_type, model_class in plugin_manager.unit_models.items():
+        unit_key_index = {'fields': model_class.unit_key_fields, 'unique': True}
+        for index in model_class._meta['indexes']:
+            if unit_key_index == index:
+                raise Exception(_("Content unit type '%s' explicitly defines an index for its unit"
```

Comment:
The docs are epicly out of date, but this could be a platform release note for plugin writers. There is usually a section for them in the release notes. +1 to putting it there